modules/cf-core/test/contracts.ts

```diff
@@ -0,0 +1,163 @@
+  DolphinCoin: string;
+  UnidirectionalTransferApp: string;
+  UnidirectionalLinkedTransferApp: string;
+  SimpleTransferApp: string;
+};
+
+export const deployTestArtifactsToChain = async (wallet: Wallet): Promise<any> => {
+  const coinBalanceRefundContract = await new ContractFactory(
+    CoinBalanceRefundApp.abi,
+    CoinBalanceRefundApp.evm.bytecode,
+    wallet,
+  ).deploy();
+
+  const dolphinCoin = await new ContractFactory(
+    DolphinCoin.abi,
+    DolphinCoin.evm.bytecode,
+    wallet,
+  ).deploy();
+
+  const identityApp = await new ContractFactory(
+    IdentityApp.abi,
+    IdentityApp.evm.bytecode,
+    wallet,
+  ).deploy();
+
+  const mvmContract = await new ContractFactory(
+    MinimumViableMultisig.abi,
+    MinimumViableMultisig.evm.bytecode,
+    wallet,
+  ).deploy();
+
+  const proxyFactoryContract = await new ContractFactory(
+    ProxyFactory.abi,
+    ProxyFactory.evm.bytecode,
+    wallet,
+  ).deploy();
+
+  const coinTransferETHInterpreter = await new ContractFactory(
+    MultiAssetMultiPartyCoinTransferInterpreter.abi,
+    MultiAssetMultiPartyCoinTransferInterpreter.evm.bytecode,
+    wallet,
+  ).deploy();
+
+  const twoPartyFixedOutcomeInterpreter = await new ContractFactory(
+    TwoPartyFixedOutcomeInterpreter.abi,
+    TwoPartyFixedOutcomeInterpreter.evm.bytecode,
+    wallet,
+  ).deploy();
+
+  const challengeRegistry = await new ContractFactory(
+    ChallengeRegistry.abi,
+    ChallengeRegistry.evm.bytecode,
+    wallet,
+  ).deploy();
+
+  const conditionalTransactionDelegateTarget = await new ContractFactory(
+    ConditionalTransactionDelegateTarget.abi,
+    ConditionalTransactionDelegateTarget.evm.bytecode,
+    wallet,
+  ).deploy();
+
+  const twoPartyFixedOutcomeFromVirtualAppETHInterpreter = await new ContractFactory(
+    TwoPartyFixedOutcomeFromVirtualAppInterpreter.abi,
+    TwoPartyFixedOutcomeFromVirtualAppInterpreter.evm.bytecode,
+    wallet,
+  ).deploy();
+
+  const tttContract = await new ContractFactory(
+    TicTacToeApp.abi,
+    TicTacToeApp.evm.bytecode,
+    wallet,
+  ).deploy();
+
+  const transferContract = await new ContractFactory(
+    UnidirectionalTransferApp.abi,
+    UnidirectionalTransferApp.evm.bytecode,
+    wallet,
+  ).deploy();
+
+  const simpleTransferContract = await new ContractFactory(
+    SimpleTransferApp.abi,
+    SimpleTransferApp.evm.bytecode,
+    wallet,
+  ).deploy();
+
+  const linkContract = await new ContractFactory(
+    UnidirectionalLinkedTransferApp.abi,
+    UnidirectionalLinkedTransferApp.evm.bytecode,
+    wallet,
+  ).deploy();
+
+  const timeLockedPassThrough = await new ContractFactory(
+    TimeLockedPassThrough.abi,
+    TimeLockedPassThrough.evm.bytecode,
+    wallet,
+  ).deploy();
+
+  const singleAssetTwoPartyCoinTransferInterpreter = await new ContractFactory(
+    SingleAssetTwoPartyCoinTransferInterpreter.abi,
+    SingleAssetTwoPartyCoinTransferInterpreter.evm.bytecode,
+    wallet,
+  ).deploy();
+
+  return {
```

Comment:
where does the provider get added in the tests now?

Reply:
Tests get an `ETHPROVIDER_URL` env var now
But tests were already using: `global["ganacheUrl"] = env.ETHPROVIDER_URL;` that earlier pointed to the local ganache so not much changed outside of the test setup.

Reply:
Tests already used `global["ganacheUrl"]` to setup the provider. In global-setup we point to an external ganache instead of the local one they used to use but besides that nothing else in tests really needed to change.